presto-native-execution/presto_cpp/main/http/HttpClient.cpp

```diff
@@ -462,10 +462,14 @@ folly::SemiFuture<proxygen::HTTPTransaction*> HttpClient::createTransaction(
     }
     VLOG(3) << "Reuse idle connection from different thread to "
             << address_.describe();
+// Skip this check for GCC to prevent SIGSEGV as described in the issue:
+// https://github.com/prestodb/presto/issues/22995
+#if !defined(__GNUC__) || defined(__clang__)
```

Comment:
Do we need `|| defined(__clang__)`? Officially we don't support other compilers than these two.

Reply:
We should just do `defined(__clang__)` then.
Clang defines GNUC also
https://stackoverflow.com/questions/38499462/how-to-tell-clang-to-stop-pretending-to-be-other-compilers